### ISSUE TITLE
Color arithmetic suggestion

### ISSUE THREAD

**Issue by ed022** [pygame/pygame#4371]
I noticed that Color arithmetic doesn't work as I, personally, would expect. I'm wondering if there is a reason for this, or if this should be corrected.

**Current behavior:**
pygame.Color(32,32,32) * pygame.Color(2,2,2) -> (64, 64, 64, 255)
pygame.Color(128,128,128) * pygame.Color(64,64,64) -> (255, 255, 255, 255)
This is what makes sense on a surface level, as it's merely multiplying the respective components together, capping it at 255 of course.

**Expected behavior:**
pygame.Color(128,128,128) * pygame.Color(64,64,64) -> (32, 32, 32, 255)
From my own experience, in practical applications, these colors should multiply as though on a scale between 0-1. That is, 128/255 * 64/255 ~= 32/255. This is how Photoshop, for example, does it too.
I suppose the converse would also hold; division should behave this way too.

Maybe I'm overlooking something; is there a practical case for direct component-component multiplication?

**Comment by LuckyIntel** [comment]
This is not a bug i think.Aren't colors a type of vector too?In some parts you can even use vectors to set colors in pygame.I really can't seem to have a fix for this but i think you can do a function like this

```py
def  blendColors(col1, col2): return pygame.Vector3(col1* col2 / 255) / 255

newColor = blendColors(pygame.Vector3(128), pygame.Vector3(64))
```

wrote this code with just github so there can be some mistakes, sorry.Made the function with the equation you gave in your issue.

**Comment by ed022** [comment]
You are correct; this is not a bug per se. I'm not sure how to give it a different tag (or, honestly, whether this is even the right place to make such a suggestion in the first place).
Your workaround is a good one; I didn't know Vectors could be use in place of Colors, so that could ease what I'm trying to do. My only annoyance is the need to use /255 frequently, which is what I propose Color should do implicitly when multiplying or dividing.
Another method that might be helpful in this context would be a conversion between Color and Vector3, which would then lend itself to such arithmetic. That is:
```
c1 = pygame.Color(255, 192, 0)
v1 = Vector3(c1)
# v1 is now (255, 192, 0)
blended = pygame.Color(v1 * v2 / 255)
```
or, even better,
```
c1 = pygame.Color(255, 192, 0)
v1 = Vector3(c1)
# v1 is now (1.0, 0.75, 0.0)
blended = pygame.Color(v1 * v2)
# blended is now something like (127, 96, 0), having automatically converted back to int space
```
Once again, the main reason I suggest this is because it caters to how - as far as I know - color arithmetic works in all practical applications. Feel free to correct me if I'm wrong about that.

**Comment by LuckyIntel** [comment]
> Feel free to correct me if I'm wrong about that.

I don't know both python and pygame that good.Let's say i am also trying to learn it.

About using / 255 so much, i really can't help it.It's the equation i need to follow.I am unsure if there is any better way to do it.

**Comment by ed022** [comment]
And that's why I suggest having it be an intrinsic behavior of Color multiplication. It's not a pygame- or Python-specific idea; this is how multiplication of Colors happens in other settings, like on GPUs and in Photoshop, for example.
I'm wondering if this simplification could also speed up the computations, because graphics processing (within software) is quite a bottleneck.

**Comment by LuckyIntel** [comment]
Time to move on to a graphics api then.Shaders will help you.

```glsl
vec4 c1 = vec4(0.5, 0.5, 0.5, 1.0); // rgb(128, 128, 128, 1)
vec4 c2 = vec4(0.25, 0.25, 0.25, 1.0); // rgb(64, 64, 64, 1)

vec4 result = c1 * c2; // this will return a vec4(0.125, 0.125, 0.125, 1.0) which is approx rgb(32, 32, 32, 1)
```

Photoshop and other graphics programs use this to multiply 2 colors.

**Comment by Osaidgit** [comment]
I run your code on my virtual env Igot no error. The problem is your pygame version is very old.Update your pip

**Comment by LuckyIntel** [comment]
@Osaidgit There seems to be no problems here.The user seems to be complaining about how multiplying/blending colors are not working as they expected.Already dropped a workaround in the replies, you can also check it or try to improve it.

Also if this problem is solved, i recommend that we should mark as solved this issue.
